ludwig/data/preprocessing.py

```diff
@@ -940,7 +940,30 @@ def build_dataset(
 
 def build_metadata(dataset_df, features, global_preprocessing_parameters):
     metadata = {}
+    already_processed_features_names = set()
+    already_processed_features_preproc_params = {}
+
     for feature in features:
+
+        # if a feature has the same name of a previously processed one
```

Comment:
An alternative way to think about this that might simplify the code would to construct a set of `(feature_name, preprocessing_params)` (to deduplicate), then only work with this set for the remainder of the preprocessing.